components/serialization/json/src/main/java/com/microsoft/kiota/serialization/JsonParseNode.java

```diff
@@ -348,15 +271,10 @@ private <T extends Parsable> void assignFieldValues(
                 final JsonElement fieldValue = fieldEntry.getValue();
                 if (fieldValue.isJsonNull()) continue;
                 if (fieldDeserializer != null) {
-                    final Consumer<Parsable> onBefore = this.onBeforeAssignFieldValues;
-                    final Consumer<Parsable> onAfter = this.onAfterAssignFieldValues;
-                    fieldDeserializer.accept(
-                            new JsonParseNode(fieldValue) {
-                                {
-                                    this.setOnBeforeAssignFieldValues(onBefore);
-                                    this.setOnAfterAssignFieldValues(onAfter);
-                                }
-                            });
+                    JsonParseNode itemNode = new JsonParseNode(fieldValue);
```

Comment:
```suggestion
                    final JsonParseNode itemNode = new JsonParseNode(fieldValue);
```

Reply:
I accept those suggestions, but I appreciate that this is a style coming strongly from dotnet.
In java codebases, I have not seen much declaring `final` in function bodies, it doesn't hurt though.